src/interface/orthnull.jl

```diff
@@ -1,220 +1,549 @@
-If NamedTuples are passed as `alg_qr`, `alg_polar`, or `alg_svd`, a default algorithm is chosen
-with `select_algorithm` and the NamedTuple is passed as keyword arguments to that algorithm.
-`alg_qr` defaults to `(; positive=true)` so that by default a positive QR decomposition will
-be used.
+    left_orth(A; [alg], [trunc], kwargs...) -> V, C
+    left_orth!(A, [VC], [alg]; [trunc], kwargs...) -> V, C
+
+Compute an orthonormal basis `V` for the image of the matrix `A`, as well as a  matrix `C`
+(the corestriction) such that `A` factors as `A = V * C`.
+
+This is a high-level wrapper where the keyword arguments can be used to specify and control
+the specific orthogonal decomposition that should be used to factor `A`, whereas `trunc`
+can optionally be used to control the precision in determining the rank of `A`, typically
+via its singular values.
+
+## Truncation
+The optional truncation strategy can be controlled via the `trunc` keyword argument, and
+any non-trivial strategy typically requires an SVD-based decompositions. This keyword can
+be either a `NamedTuple` or a [`TruncationStrategy`](@ref).
+
+### `trunc::NamedTuple`
+The supported truncation keyword arguments are:
+
+$(docs_truncation_kwargs)
+
+### `trunc::TruncationStrategy`
+For more control, a truncation strategy can be supplied directly.
+By default, MatrixAlgebraKit supplies the following:
+
+$(docs_truncation_strategies)
+
+## Keyword arguments
+There are 3 major modes of operation, based on the `alg` keyword, with slightly different
+application purposes.
+
+### `alg::Nothing`
+This default mode uses the presence of a truncation strategy `trunc` to determine an optimal
+decomposition type, which will typically be QR-based for no truncation, or SVD-based for
+truncation. The remaining keyword arguments are passed on directly to the algorithm selection
+procedure of the chosen decomposition type.
+
+### `alg::Symbol`
+Here, the driving selector is `alg`, which is used to select the kind of decomposition. The
+remaining keyword arguments are passed on directly to the algorithm selection procedure of
+the chosen decomposition type. By default, the supported kinds are:
+
+* `:qr` : Factorize via QR decomposition, with further customizations through the other
+  keywords. This mode requires `isnothing(trunc)`, and is equivalent to
+```julia
+        V, C = qr_compact(A; kwargs...)
+```
+
+* `:polar` : Factorize via polar decomposition, with further customizations through the
+  other keywords. This mode requires `isnothing(trunc)`, and is equivalent to
+```julia
+        V, C = left_polar(A; kwargs...)
+```
+
+* `:svd` : Factorize via SVD, with further customizations through the other keywords.
+  This mode further allows truncation, which can be selected through the `trunc` argument,
+  and is roughly equivalent to:
+```julia
+        V, S, C = svd_trunc(A; trunc, kwargs...)
+        C = S * C
```

Comment:
I find this way of writing it a bit confusing. What about:
```suggestion
        V, S, Uᴴ = svd_trunc(A; trunc, kwargs...)
        C = S * Uᴴ
```
?